radis/spectrum/spectrum.py

```diff
@@ -2292,55 +2395,92 @@ def apply_slit(
                     verbose=verbose,
                     assert_evenly_spaced=False,
                     # assumes Spectrum is correct by construction
-                    **kwargsconvolve
+                    **kwargsconvolve,
                 )
 
                 if i == 0:
                     w_conv_slices.append(w_conv_window)
                 I_conv_slices[q].append(I_conv_window)
 
-        # Merge and store all variables
-        # ---------
+        # Get units
+        new_units = {}
         for q in I_conv_slices.keys():
             qns = q + "_noslit"
 
-            # Merge all slices
-            w_conv = np.hstack(w_conv_slices)
-            I_conv = np.hstack(I_conv_slices[q])
-
-            # Store
-            self._q_conv["wavespace"] = w_conv
-            self._q_conv[q] = I_conv
-
             # Get units
             if norm_by == "area":
-                self.units[q] = self.units[qns]
+                new_units[q] = self.units[qns]
             elif norm_by == "max":
                 new_unit = (Unit(unit) * Unit(self.units[qns])).to_string()
                 # because it's like if we multiplied by slit FWHM in the wavespace
                 # it was generated
-                self.units[q] = new_unit
+                new_units[q] = new_unit
             # Note: there was another mode called 'max2' where, unlike 'max',
             # unit was multiplied by [unit] not [return_unit]
             # Removed for simplification. You should stay with norm_by='area' anyway
             else:
                 raise ValueError("Unknown normalization type: {0}".format(norm_by))
 
+        # Merge and store all variables
+        # ---------
+        if inplace:
+            w_conv = np.hstack(w_conv_slices)
+            if len(self._q["wavespace"]) != len(w_conv) or not np.allclose(
+                self._q["wavespace"], w_conv
+            ):
+                raise AssertionError(
+                    "Wavespace of convolved arrays is different, cannot store it in the same Spectrum. You can use Spectrum.apply_slit(inplace=False) to return a new spectrum with only the convolved arrays"
+                )
+            for q in I_conv_slices.keys():
+                qns = q + "_noslit"
+
+                # Merge all slices
+                I_conv = np.hstack(I_conv_slices[q])
+
+                # Store
+                self._q[q] = I_conv
+
+                # Get units
+                self.units[q] = new_units[q]
+            s_out = self
+        else:
+            w_conv = np.hstack(w_conv_slices)
+            # make new spectrum
+            quantities = I_conv_slices.items()
+            quantities = {
+                "wavespace": w_conv
+            }  # a copy will be created in Spectrum creation
+
+            for q in I_conv_slices.keys():
+
+                # Merge all slices
+                quantities[q] = np.hstack(I_conv_slices[q])
```

Comment:
this part too

Reply:
No, this one is where we do the stacking. 
Instead I remove 
```
            quantities = I_conv_slices.items()
```
from the line above